src/meta/src/hummock/compaction_scheduler.rs

```diff
@@ -91,76 +93,79 @@ where
-            None => {
-                tracing::warn!("No compactor is available.");
-                return;
-            }
-            Some(compactor) => compactor,
-        };
-
-        // 2. Pick a compact task and assign to the compactor.
+    async fn pick_and_assign(&self, _compaction_group: CompactionGroupId) {
+        // 1. Pick a compaction task.
         // TODO: specify compaction_group in get_compact_task
-        let mut compact_task = match self
-            .hummock_manager
-            .get_compact_task(compactor.context_id())
-            .await
-        {
+        let compact_task = match self.hummock_manager.get_compact_task().await {
             Ok(Some(compact_task)) => compact_task,
             Ok(None) => {
-                // No compact task available.
+                // No compaction task available.
                 return;
             }
             Err(err) => {
-                tracing::warn!(
-                    "Failed to get compact task for compactor {}. {:#?}",
-                    compactor.context_id(),
-                    err
-                );
-                self.compactor_manager
-                    .remove_compactor(compactor.context_id());
-                self.reschedule_compaction_group(compaction_group);
+                tracing::warn!("Failed to get compaction task: {:#?}.", err);
                 return;
             }
         };
+        tracing::info!(
+            "Picked compaction task. {}",
+            compact_task_to_string(&compact_task)
+        );
 
-        // 3. Send the compact task to the compactor.
-        match compactor.send_task(Some(compact_task.clone()), None).await {
-            Ok(_) => {
-                tracing::debug!(
-                    "Sent compaction task {} to worker {}.",
-                    compact_task_to_string(&compact_task),
-                    compactor.context_id()
-                );
-                // TODO: decide if more compaction task available in compaction_group, then either
-                // reschedule or unset compaction_group's is_scheduled.
-            }
-            Err(err) => {
-                tracing::warn!(
-                    "Failed to send compaction task to worker {}: {}",
-                    compactor.context_id(),
-                    err
-                );
-                compact_task.task_status = false;
-                if let Err(err) = self
-                    .hummock_manager
-                    .report_compact_task(&compact_task)
-                    .await
-                {
+        // 2. Assign the compaction task to a compactor.
+        'send_task: loop {
+            // 2.1 Select a compactor.
+            let compactor = match self.compactor_manager.next_compactor() {
+                None => {
+                    tracing::warn!("No compactor is available.");
+                    tokio::time::sleep(Duration::from_secs(60)).await;
```

Comment:
Will `60s` be too  long? In my experience, a compaction job shall complete in 1s ~ 30s.